src/jdk.jdwp.agent/share/native/libjdwp/eventHandler.c

```diff
@@ -1239,6 +1239,10 @@ cbVMDeath(jvmtiEnv *jvmti_env, JNIEnv *env)
         EXIT_ERROR(error,"Can't clear event callbacks on vm death");
     }
 
+    /* Setting this flag is needed for a very special case. See the reference in findThread(). */
+    gdata->handlingVMDeath = JNI_TRUE;
```

Comment:
Is this used to communicate across different threads? If so then it needs to be at least volatile, but a more forceful memory-barrier may be needed on some platforms. I can't tell in what context findThread will be called after this has been set to true.

Reply:
I had an answer typed up for you, but I realized I need to think some things through a bit more, and it's getting late. I'll look into it some more tomorrow.